README.md

```diff
@@ -1,10 +1,47 @@
-## Dock DID Auth
+# Dock Web3 ID
 
-This service provides an oauth solution for authenticating with DIDs through Verifiable Credentials.
+This service provides a SSI-friendly authentication solution for use with DIDs and Verifiable Credentials. There is an existing live service hosted at https://auth.dock.io however you may clone this repository/deploy it to vercel to spin up your own instance. It is mostly stateless, but does rely on a memcached instance currently for short-lived data transfer. No long term storage of user data is used, tokens and codes are obtained through cryptography. You can use this service to allow your users to provide their own user data, like you would request from "Login with Facebook" or "Sign in with Github".
 
-## Getting Started
+Note: You will still want to verify a users email if requested with this service.
 
-First, setup [#env-vars](the environment variables) and pre-requisite services and then you can run the development server:
+Features:
+- No long term storage of user/client data
+- User provides and controls their own data
+- Uses `did:dock` and `did:key` DIDs
+- Cryptographic client id/secrets
+- Oauth2 spec compliant
+
+Roadmap:
+- Support requesting specific credentials/data
+- Decouple verification from the Dock API
+- Look into supporting the SIOP spec
+- OpenID Connect implementation
+- Support more DID types
+- Support non self-signed credentials
+
+## As an oauth2 provider
+
+This service can be used directly as an oauth2 provider with your favourite oauth library. See the documentation for [Oauth2 setup](docs/oauth2_setup.md). You are welcome to use our hosted version or your own. Setup instructions are for the hosted vesion, simply replace with your own domain to configure for another endpoint.
+
+## As an Auth0 connection
+
+This service can be used as an Auth0 Social Connector allowing users to sign into applications using their DIDs. See [Auth0 Application Setup](docs/auth0_configure_application.md)
+
+## Under the Hood
+
+Decentralized Identifiers (DIDs) are cryptographically verifiable pseudonymous identifiers created by the user, owned by the user, and independent of any organization. DIDs contain no personal data about the user, the user may provide extra data you request such as their name, email etc. An example of a DID stored on the Dock blockchain could look like this:
+
+![sample-did](./public/DID%20example.jpg)
+
+Each DID is supported by a Public-Private cryptographic key pair.
+
+When a user scans the QR Code generated by the Dock Web3 ID Auth0 integration, the user’s Private Key associated with the DID digitally signs a Verifiable Credential. This Verifiable Credential contains a cryptographic hash that ensures that it wasn’t modified since it was created and signed - and most importantly verifies that the user who owns that DID is providing that data.
```

Comment:
remove Auth0 reference here - we probably don't need it even when we do bring Auth0 integration out.

Reply:
see other comment